wxMSW: Try to fix wxTextCtrl::AppendText() for plain EDIT controls

AliKet opened the pull request:
See #25164

For RICHEDIT controls the maxlimit is updated after each `AppendText()` calls and don't know how to fix it?
maybe we should just document this limitation and live with it as is ?

aliket83 committed:
wxMSW: Fix wxTextCtrl::AppendText() which fails if max length is set

- If explicitly set, save/restore the old max limit using bitwise complement.

aliket83 committed:
Enable TextCtrlTestCase::LongText() test to run under all platforms

And optimize it by just calling AppendText() once, which greatly reduces execution time.

SetMaxLength() is removed and will be tested thoroughly in the upcoming commit.

aliket83 committed:
Add TextCtrlTestCase::MakeLinePattern() helper function

No real changes, just a refactoring as this function will be used
in the upcoming commit.

aliket83 committed:
Update TextCtrlTestCase::MaxLength() to test SetMaxLength() for multiline control

vadz reviewed (commented):
Thanks, I _think_ this does the right thing, but this code is really difficult to follow.

Maybe we could change `gs_lenOfInsertedText` to be a `stack<InsertTextData>` where `InsertTextData` would be a struct with separate fields for the length of the text being inserted and the old max length, which would be set only if it was changed? I.e. use 2 separate variables instead of interpreting a single int differently depending on its sign?

aliket83 committed:
Rename gs_lenOfInsertedText to gs_insertTextData

No real change in functionality, just make the variable a wxStack<InsertTextData>
instead of wxStack<int> to make the code clearer and easier to understand.

AliKet commented on src/msw/textctrl.cpp:
```diff
@@ -1929,9 +1931,9 @@ wxString wxTextCtrl::GetLineText(long lineNo) const
 {
 #if wxUSE_RICHEDIT
-    if ( IsRich() )
+    if ( IsRich() && len > 64000 )
```
I just restored the old code

AliKet commented on src/msw/textentry.cpp:
```diff
@@ -968,7 +968,7 @@ void wxTextEntry::SetMaxLength(unsigned long len)
     }
 
-    ::SendMessage(GetEditHwnd(), EM_LIMITTEXT, len, 0);
+    ::SendMessage(GetEditHwnd(), EM_SETLIMITTEXT, len, 0);
```
Restored the old name.

aliket83 committed:
wxQt: Implement maximum length for the multiline text control

vadz commented:
Looks good to me, thanks! Will merge soon.

vadz commented:
Oops, no, sorry, there is an error in one of AppVeyor jobs.
